app/discovery/provider/consulcatalog/client.go

```diff
@@ -0,0 +1,147 @@
+package consulcatalog
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"sort"
+	"strings"
+	"time"
+)
+
+type Client struct {
+	address string
+	client  *http.Client
+}
+
+func NewClient(address string, timeout time.Duration) *Client {
```

Comment:
I'd suggest passing http.Clinet in. IT will be much more flexible. For instance we could levarage a clinet with a custom transport implementing retries